examples/dreambooth/train_dreambooth.py

```diff
@@ -811,12 +846,63 @@ def main(args):
+                            # Load current training state into a new diffusion pipeline to generate samples
+                            pipeline = DiffusionPipeline.from_pretrained(
+                                args.pretrained_model_name_or_path,
+                                unet=accelerator.unwrap_model(unet),
+                                text_encoder=accelerator.unwrap_model(text_encoder),
+                                torch_dtype=weight_dtype
+                            ).to(accelerator.device)
+
+                            if args.convert_checkpoints:
+                                convertedOutPath = os.path.join(args.output_dir, "converted_checkpoints", f"checkpoint-{global_step}")
+                                logger.info(f"Converting checkpoint-{global_step} to diffusers model at {convertedOutPath}")
+                                pipeline.save_pretrained(convertedOutPath)
+
+                            logger.info(f"Generating {args.samples_per_checkpoint} samples at step {global_step} with prompt: {args.sample_prompt}")
+
+                            # Allow a statically set or random seed for for generated samples
+                            sampleSeed = args.sample_seed if args.sample_seed != -1 else torch.Generator(accelerator.device).seed()
+                            sampleGenerator = torch.Generator(device=accelerator.device)
+
+                            for sampleIdx in range(args.samples_per_checkpoint):
+                                sampleGenerator.manual_seed(sampleSeed)
+                                # Generate samples
+                                with torch.cuda.amp.autocast(enabled=True):
+                                    images = pipeline(
+                                        prompt=args.sample_prompt, 
+                                        num_images_per_prompt=1,
+                                        num_inference_steps=args.sample_steps,
+                                        generator=sampleGenerator,
+                                        width=args.resolution,
+                                        height=args.resolution).images
+
+                                # Save samples to 'samples' folder in output directory
+                                for i, image in enumerate(images):
+                                    hash_image = hashlib.sha1(image.tobytes()).hexdigest()
+                                    os.makedirs(os.path.join(args.output_dir, "samples"), exist_ok=True)
+                                    image_filename = os.path.join(args.output_dir, "samples", f"step-{global_step}_seed-{sampleSeed}_loss-{loss.detach().item()}_hash-{hash_image}.png")
+                                    image.save(image_filename)
+                                
+                                sampleSeed += 1
+                            
+                            # Remove interim pipeline reference
+                            del pipeline
```

Comment:
This can be simplified a lot , we have an example for logging, cf https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/train_dreambooth_lora.py#L902

Reply:
Hey @subpanic, could you maybe try to use the mechanism of https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/train_dreambooth_lora.py#L902 here for generation? It should be a bit easier than what we currently have :-)